app/app.go

```diff
@@ -269,13 +269,6 @@ func NewInitApp(
 
 		staking.NewAppModule(app.stakingKeeper, app.accountKeeper, app.supplyKeeper),
 	)
-	// During begin block slashing happens after distr.BeginBlocker so that
-	// there is nothing left over in the validator fee pool, so as to keep the
-	// CanWithdrawInvariant invariant.
-
-	app.mm.SetOrderBeginBlockers(distr.ModuleName, slashing.ModuleName)
```

Comment:
This is deleted because `module.NewManager` is already creating ordering this based on the order of the module we give.

Reply:
But it won't be the same order.

Reply:
Is there any specific reason why we have this order and why we remove other modules in the first place?

If you look at all the different modules, half of them are implementing the `BeginBlock` function 
- [distribution](https://github.com/cosmos/cosmos-sdk/blob/165e134405c5e4f2a1115e83321abd1328feb9c2/x/distribution/module.go#L146)
- [evidence](https://github.com/cosmos/cosmos-sdk/blob/165e134405c5e4f2a1115e83321abd1328feb9c2/x/evidence/module.go#L160)
- [mint](https://github.com/cosmos/cosmos-sdk/blob/165e134405c5e4f2a1115e83321abd1328feb9c2/x/mint/module.go#L129)
- [slashing](https://github.com/cosmos/cosmos-sdk/blob/165e134405c5e4f2a1115e83321abd1328feb9c2/x/slashing/module.go#L143)
- [staking](https://github.com/cosmos/cosmos-sdk/blob/165e134405c5e4f2a1115e83321abd1328feb9c2/x/staking/module.go#L169)
- [upgrade](https://github.com/cosmos/cosmos-sdk/blob/165e134405c5e4f2a1115e83321abd1328feb9c2/x/upgrade/module.go#L128)

Why do we only activate the distribution and slashing in the first place?

Reply:
Also, the order is the same as we add `distribution` before `slashing` in the `module.NewManager`

Reply:
code reverted for now, we can think about that later